### DISCUSSION TITLE
Server action not found when using App Route with i18n

### DISCUSSION THREAD

**Discussion by zkSoju** [amannn/next-intl#1127] [Q&A]
### Description

When I have i18n routing setup with server actions, it is giving me a 404 error. It seems to be an issue with the locales dynamic routes? I am also using `unstable_setRequestLocale` in my server component. I have tried with and without static generation and neither work.

Server action also works locally but not when deployed

<img width="1086" alt="Screenshot 2024-06-13 at 2 14 19 AM" src="https://github.com/amannn/next-intl/assets/28630104/6d99c34d-270d-4da5-a797-9fc5af0eef31">

### Verifications

- [X] I've verified that the problem I'm experiencing isn't covered in [the docs](https://next-intl-docs.vercel.app/).
- [X] I've searched for similar, existing issues on GitHub and [Stack Overflow](https://stackoverflow.com/search?q=next-intl).
- [X] I've compared my app to [a working example](https://next-intl-docs.vercel.app/examples) to look for differences.

### Mandatory reproduction URL

https://github.com/0xHoneyJar/minimal-reprod

### Reproduction description

Steps to reproduce:
1. Open reproduction
2. Deploy to vercel
3. Click server action button in top left corner

### Expected behaviour

The server actions should be found and operate normally.

**Comment by amannn** [comment]
I'm getting a 200 in your deployed example:

![Screenshot 2024-06-14 at 08 29 46](https://github.com/amannn/next-intl/assets/4038316/d4f88a79-57e2-453c-bd5f-c340838699e7)

Did you fix something after posting this?

If a 404 is triggered, it's typically caused by the middleware not matching a given request (see "[Unable to find next-intl locale because the middleware didn't run on this request.](https://next-intl-docs.vercel.app/docs/routing/middleware#unable-to-find-locale)").

**Comment by amannn** [comment]
I'll move this to a discussion since it seems to be a usage question and not a bug report, esp. since you mention that it works locally.
